Chemical Flamethrowers and Reagent Explosion Fix

KoboldCommando opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

First off, a fix that was relevant: when an explosive chem reaction occurs (namely black powder), it will now actively track if the holder is moved or deleted, meaning that black powder grenades will now EXPLODE rather than turning into a wet fart and crashing!

The bulk of this PR is more firemodes for flamethrowers!

-fire your flamethrower without ignition to spray gas without flames!
-load your flamethrower with a beaker full of chemicals!
-fire your ignited flamethrower with chemicals, to heat up the chemicals, potentially causing them to react after firing!

Also adds sprites for having a beaker loaded, and inhands so it can show up.

I don't think these files can really be modularized.

## Why It's Good For The Game

This is going to enable an IMMENSE amount of shenanigans.  Helpful, deadly, or destructive.  Explosion throwers, medicine throwers, firefighting sprayers, repressurization gun, A DEEP FRY CANNON, who knows what people will come up with!

<summary>Screenshots&Videos</summary>

https://cdn.discordapp.com/attachments/908934283513696267/940038333889937448/Chem_Thrower.mp4

https://cdn.discordapp.com/attachments/908934283513696267/940047989253214208/2022-02-06_19-53-50.mp4


## Changelog
:cl:
add: you can now load a beaker of chemicals into flamethrowers, and also fire them with or without the flame lit!
fix: black powder grenades and similar explosive reactions work properly now!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

nednaZ commented on code/game/objects/items/flamethrower.dm:
```diff
@@ -25,6 +25,7 @@
 	var/obj/item/weldingtool/weldtool = null
 	var/obj/item/assembly/igniter/igniter = null
 	var/obj/item/tank/internals/plasma/ptank = null
+	var/obj/item/reagent_containers/glass/bker = null
```
The description of the flamethrower should probably be updated to say that you can attach a beaker to it.
Also the variable name would likely be better as "beaker" to be more descriptive.

nednaZ reviewed (commented):
As it stands, a 15u beaker can spray over 40 times before it just turns into generic mist.
I'd say it needs a "minimum" amount for each spray, maybe 5u a shot.

KoboldCommando commented:
That's a good point, and partly a bug.  I doubled the draw from the beaker, without increasing the application.  So now a large beaker will apply 5u to each tile, while draining 10u from the beaker.  This seemed pretty reasonable to me.

I also added checks so that it won't make sound or try to fire when empty.

I also added a check to the tank so that you need to have more gas pressure in the tank than there is in the turf you're firing from to actually fire, which puts a lower end on how much you can fire and prevents having a dozen pitiful little squirts at the end.
